plugins/routeros-suspension/src/src/Service/UnmsApi.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace UcrmRouterOs\Service;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Response;
+use Ubnt\UcrmPluginSdk\Exception\ConfigurationException;
+use Ubnt\UcrmPluginSdk\Service\PluginConfigManager;
+use Ubnt\UcrmPluginSdk\Service\UcrmOptionsManager;
+use Ubnt\UcrmPluginSdk\Util\Helpers;
+use Ubnt\UcrmPluginSdk\Util\Json;
+
+class UnmsApi
+{
+    private const HEADER_AUTH_TOKEN = 'x-auth-token';
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $token;
+
+    public function __construct(Client $client, string $token)
+    {
+        $this->client = $client;
+        $this->token = $token;
+    }
+
+    public static function create(): self
+    {
+        $options = (new UcrmOptionsManager())->loadOptions();
+
+        $ucrmUrl = ($options->ucrmLocalUrl ?: $options->ucrmPublicUrl) ?? '';
+        if (! $ucrmUrl) {
+            throw new ConfigurationException('UCRM URL is missing in plugin configuration.');
+        }
+
+        $config = (new PluginConfigManager())->loadConfig();
+        if (! ($config['unmsApiToken'] ?? false)) {
+            throw new ConfigurationException('UNMS API token is missing in plugin configuration.');
+        }
+
+        $unmsApiUrl = sprintf('%s/api/v2.1/', str_replace('crm/', 'nms', $ucrmUrl));
```

Comment:
This looks like a bug, even though it probably isn't. Can we maybe do something like this?
```suggestion
        $unmsApiUrl = sprintf('%s/api/v2.1/', rtrim(str_replace('crm/', 'nms/', $ucrmUrl), '/'));
```